h/views/activity.py

```diff
@@ -42,9 +42,81 @@ def search(request):
     # Fetch results
     result = query.execute(request, q, page_size=page_size)
 
+    for user in result.aggregations.get('users', []):
+        user['username'] = util.user.split_user(user['user'])['username']
+        del user['user']
+
     return {
         'total': result.total,
         'aggregations': result.aggregations,
         'timeframes': result.timeframes,
         'page': paginate(request, result.total, page_size=page_size),
     }
+
+
+@view_config(route_name='activity.group_search',
+             request_method='GET',
+             renderer='h:templates/activity/search.html.jinja2')
+def group_search(request):
+    if not request.feature('search_page'):
+        raise httpexceptions.HTTPNotFound()
+
+    result = search(request)
+
+    pubid = request.matchdict['pubid']
+
+    try:
+        group = request.db.query(models.Group).filter_by(pubid=pubid).one()
+    except exc.NoResultFound:
+        return result
+
+    if request.authenticated_user not in group.members:
+        return result
+
+    result['group'] = {
+        'created': group.created.strftime('%B, %Y'),
+        'description': group.description,
+        'name': group.name,
+        'pubid': group.pubid,
+    }
+
+    result['show_group_leave_button'] = (
+        group.creator != request.authenticated_user)
```

Comment:
Another thing that I chose not to fix right now because the details seems likely to be controversial and I'm in a bit of a hurry: it might be nicer if the Pyramid permissions system can tell us whether the authenticated user has "permission" to leave the group (as it tells us whether they have permission to "admin" the group below), rather than encoding the "group creators cannot leave the group (but members can)+ business logic here in the view function.